### PR TITLE
Remove for..of, await in loop eslint warnings

### PR DESCRIPTION
**Pull request by chrischrischris**
We use await inside of for..of loops enough that we don't need to have an eslint warning for it.  Saves a bit of file size too.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -10,26 +10,37 @@ module.exports = {
...
-    'no-unused-expressions': 0,
-    'chai-friendly/no-unused-expressions': 2,
-
   },
   overrides: [
-    {
-      files: ['test/**/*.js'],
-      rules: { 'no-console': 'off' },
-    },
+    { files: ['test/**/*.js'] },
```
Comment (overmyheadandbody) [inline .eslintrc.js:43]: Since there's no longer an override for this path, is it worth keeping it?

**Review by overmyheadandbody — APPROVED** [review]
Good idea, no use in having rules that just make developers mute them in so many cases.

**Review by sukamat — APPROVED** [review]

**Review by mokimo — APPROVED** [review]
Cool like the initiative of customizing the eslint rules as it fits the code base based on how we actually write code. 👍

File: .eslintrc.js (hunk 1 above)
Comment (chrischrischris) [reply .eslintrc.js:43]: eh probably not but I'm going to leave it just in case at some point we want to add other test folders

**Pull request merged**